### PR TITLE
Support injecting headers

### PR DESCRIPTION
**Pull request by samiura**
**Description:**  Injecting headers 
**Link to tracking Issue:**#33260

**Testing:** Tested locally using Kafka -> Kafka receiver -> otel-collector -> kafka exporter

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

**Comment by samiura** [comment]
I will fix the tests.

**Comment by crobert-1** [comment]
> I am looking at the e2e k8s test failures

There appears to be some flakiness on `main` today with the [`e2e-tests` action](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/workflows/e2e-tests.yml?query=branch%3Amain), I don't believe those failures are related to this PR.

**Review by MovieStoreGuy — COMMENTED** [review]
This only adds it into the configuration, is there work to apply it to the client?

File: exporter/kafkaexporter/config.go [hunk 1]
```diff
@@ -64,6 +67,14 @@ type Config struct {
 	Authentication kafka.Authentication `mapstructure:"auth"`
 }
 
+type HeaderInjection struct {
+	// Whether to inject headers or not, default is false.
+	InjectHeaders bool `mapstructure:"inject_headers"`
+
+	// list of header complaint with kafka receiver
```
Comment (pavolloffay) [inline exporter/kafkaexporter/config.go:74]: What are the compliant headers? How does a user know which headers are compliant?

File: exporter/kafkaexporter/config.go [hunk 2]
```diff
@@ -64,6 +67,14 @@ type Config struct {
 	Authentication kafka.Authentication `mapstructure:"auth"`
 }
 
+type HeaderInjection struct {
+	// Whether to inject headers or not, default is false.
+	InjectHeaders bool `mapstructure:"inject_headers"`
```
Comment (atoulme) [inline exporter/kafkaexporter/config.go:72]: Is this boolean needed? You might not need that, just check any headers are present.

File: exporter/kafkaexporter/config.go (hunk 1 above)
Comment (atoulme) [reply exporter/kafkaexporter/config.go:74]:
```suggestion
	// headers to inject in kafka messages
```

File: exporter/kafkaexporter/config.go [hunk 3]
```diff
@@ -64,6 +67,14 @@ type Config struct {
...
+
+	// list of header complaint with kafka receiver
+	Headers []string `mapstructure:"headers"`
```
Comment (atoulme) [inline exporter/kafkaexporter/config.go:75]: Please add code to set the headers.

**Pull request closed without merging**